pinot-core/src/main/java/org/apache/pinot/core/operator/CombineOperator.java

```diff
@@ -126,8 +126,12 @@ public void runJob() {
         }
       });
     }
+    // TODO: we can track a volatile variable "desiredLimitReached"
```

Comment:
We will cancel the future in line 187. Here we should catch the InterruptedException and silently return, or we might end up logging lots of invalid ERROR messages.